metrics/coverage_converter/docs/02_coverage_converter_publishing.md

```diff
@@ -0,0 +1,225 @@
+
+This document does not include information about configuring building jobs for the Coverage Converter tool.
+
+# Design
+
+> Explain and diagram the technical design
+
+`Coverage Converter` -> `Build using GitHub Actions` -> `GitHub Releases publishing`
+
+> Identify risks and edge cases
+
+# API
+
+> What will the proposed API look like?
+
+# Coverage Converter publishing
+
+There are a couple of places we can publish the Coverage Converter: 
+
+- GitHub Releases.
+- Ubuntu Personal Package Archive (PPA), Homebrew, and Chocolatey.
+- pub.dev publishing.
+
+
+Since we are using the GitHub Actions for building our projects, the GitHub Releases seems to be the most convenient way of publishing. Let's take a closer look at the GitHub Releases: 
+
+## GitHub Releases
+
+The GitHub provides an easy way of publishing the applications - GitHub Releases. It allows you to create a GitHub release and publish any files you want there. So, to publish a Coverage Converter CLI, we'll need to follow the next steps: 
+
+1. Build the Coverage Converter tool.
+2. Create a new release.
+3. Add executables to created release.
+
+Moreover, GitHub Actions has ready-to-use actions that simplify the process of publishing: 
+
+- [Create a release action](https://github.com/marketplace/actions/create-a-release) - an action that helps to create a new release.
+- [Upload a release asset action](https://github.com/marketplace/actions/upload-a-release-asset) - an action that helps to add assets to the existing release.
+
+### Pros & Cons
+
+Pros: 
+- Easy to configure.
+- Great integration with GitHub Actions.
+- Does not require any configuration files to publish.
+- Can publish any file type, so downloading the CLI user gets the ready-to-use application.
+
+Cons: 
+- More complex configuration for monorepo (could be solved with tags, etc..)
+- Downloading to CI action becomes a bit more complex compared to the [alternatives](#Personal-Package-Archive-(PPA)-Homebrew-and-Chocolatey), see more [here](#Downloading-using-the-command-line-commands)
+
+So, the GitHub Releases is the best choice in our case because it allows us easily configure the publishing and allows users to easily download and start using the Coverage Converter tool. 
+
+### Publishing details
+
+Once we've chosen the [GitHub Releases](#GitHub-Releases), let's consider the deployment process in more detail. To publish the application to the GitHub releases, we should create a tag that will correspond to the release. Since we want to keep the release history and we'll have more than one project on our repository, we should create a tag that will contain the version number and the name of the project - `v1.0-coverage-converter`. Also, we want to publish the `pre-release` versions, so we should create a tag for those releases. Since we don't want to keep the `pre-release` versions history, we can create a `coverage-converter-snapshot` tag that will be re-assigned on each release.
```

Comment:
```suggestion
Once we've chosen the [GitHub Releases](#GitHub-Releases), let's consider the deployment process in more detail. To publish the application to the GitHub releases, we should create a tag that will correspond to the release. Since we want to keep the release history and we'll have more than one project on our repository, we should create a tag that will contain the version number and the name of the project - `v1.0-coverage-converter`.

For every time that we change coverage converter tool sources, we use a `coverage-converter-snapshot` tag that will be re-assigned after each build to the newly published binary. It should be marked as `pre-release`. There should be only 1 release at a time with `coverage-converter-snapshot` tag.
```
Would that make sense?